Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosDatabaseAccountSettingsTests.cs

```diff
@@ -4,48 +4,90 @@
+        
+      
+        [TestMethod]
+        public async Task GetCosmosDatabaseAccountSettings_WhenEnabledClientTelemetry()
         {
-            this.cosmosClient = TestCommon.CreateCosmosClient();
-        }
+            AccountClientConfiguration clientConfig = new AccountClientConfiguration
+            {
+                ClientTelemetryConfiguration = new ClientTelemetryConfiguration
+                {
+                    IsEnabled = true,
+                    Endpoint = telemetryEndpoint
+                }
+            };
 
-        [TestCleanup]
-        public void TestCleanup()
-        {
-            if (this.cosmosClient != null)
+            HttpClientHandlerHelper handlerHelper = new HttpClientHandlerHelper
             {
-                this.cosmosClient.Dispose();
-            }
+                RequestCallBack = (request, cancellation) =>
+                {
+                    if (request.RequestUri.AbsoluteUri.Contains(Paths.ClientConfigPathSegment))
+                    {
+                        HttpResponseMessage result = new HttpResponseMessage(HttpStatusCode.OK);
+                        string payload = JsonConvert.SerializeObject(clientConfig);
+                        result.Content = new StringContent(payload, Encoding.UTF8, "application/json");
+
+                        return Task.FromResult(result);
+                    }
+                    return null;
+                }
+            };
+
+            CosmosClient cosmosClient = TestCommon.CreateCosmosClient(
+                useGateway: true,
+                customizeClientBuilder: clientBuilder => clientBuilder.WithHttpClientFactory(() => new HttpClient(handlerHelper)));
+            
+            AccountProperties accountProperties = await cosmosClient.ReadAccountAsync();
+            Assert.IsNotNull(accountProperties);
+            Assert.IsNotNull(accountProperties.Id);
+            Assert.IsNotNull(accountProperties.ReadableRegions);
+            Assert.IsTrue(accountProperties.ReadableRegions.Count() > 0);
+            Assert.IsNotNull(accountProperties.WritableRegions);
+            Assert.IsTrue(accountProperties.WritableRegions.Count() > 0);
+
+            Assert.IsNotNull(accountProperties.ClientConfiguration);
+            Assert.IsNotNull(accountProperties.ClientConfiguration.ClientTelemetryConfiguration);
+            Assert.AreEqual(telemetryEndpoint, accountProperties.ClientConfiguration.ClientTelemetryConfiguration.Endpoint);
+            Assert.IsTrue(accountProperties.ClientConfiguration.ClientTelemetryConfiguration.IsEnabled);
+
+            cosmosClient.Dispose();
         }
 
+
         [TestMethod]
-        public async Task GetCosmosDatabaseAccountSettings()
+        public async Task GetCosmosDatabaseAccountSettings_WhenDisabledClientTelemetry()
         {
-            AccountProperties accountProperties = await this.cosmosClient.ReadAccountAsync();
+            CosmosClient cosmosClient = TestCommon.CreateCosmosClient(
```

Comment:
```suggestion
            using CosmosClient cosmosClient = TestCommon.CreateCosmosClient(
```

That way if the test fails, the client is disposed anyway. Removing the dispose call